src/plugin.ts

```diff
@@ -0,0 +1,50 @@
+// global
+import { FastifyPluginAsync } from 'fastify';
+
+// local
+import { SearchService } from './db-service';
+import { TaskManager } from './task-manager';
+import { search } from './schemas';
+import { GraaspSearchPluginOptions } from './types';
+
+const plugin: FastifyPluginAsync<GraaspSearchPluginOptions> = async (fastify, options) => {
+  const {
+    taskRunner: runner
+  } = fastify;
+  const searchService = new SearchService(options.publishedTagId);
+  const taskManager = new TaskManager(searchService);
+
+  enum ranges {
+    title = 'title',
+    tag = 'tag',
+    all = 'all',
+    author = 'author'
+  }
```

Comment:
From this typescript convention it's better to capitalize the enum and its values https://www.typescriptlang.org/docs/handbook/enums.html

Reply:
Do I also need to capitalize assigned values? However if so, the Param in endpoint url would also be capitalized. Would that comply with the convention?

Reply:
No probably the strings don't need to be capitalized, but have Ranges and Title, Tag, All = 'all'